File: Languages/MATLAB/pi_monte_carlo.m
%=====================================================================
% Program: pi_monte_carlo.m
%
%          Parallel Monte Carlo calculation of PI
%
% Run:     matlab -nosplash -nodesktop -nodisplay -r "pi_monte_carlo"
%=====================================================================
R = 1.0;
darts = 1e6;
count = 0;
for i = 1:darts
  % Compute the X and Y coordinates of where the dart hit the.........
  % square using Uniform distribution.................................
  x = R*rand(1);
  y = R*rand(1);
  if x^2 + y^2 <= R^2
    % Increment the count of darts that fell inside of the circle.....
    count = count + 1; % Count is a reduction variable.
  end
end
% Compute pi..........................................................
myPI = 4*count/darts;
fprintf('The computed value of pi is %8.7f.\n',myPI);
exit;

File: Languages/Julia/Example1/pi_monte_carlo.jl
#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
# Program: pi_monte_carlo.jl
#          Monte-Carlo calculation of PI
#
# Usage: julia pi_monte_carlo.jl
#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
function montepi(n::Int)
   R = 1.0
   s = 0
   for i = 1: n
      x = R * rand()
      y = R * rand()
      if x^2 + y^2 <= R^2
         s = s + 1
      end
   end
   return 4.0*s/n
end

# Main program
for i in 3: 8
    n = 10^i
    p = montepi(n)
    println("N = $n: PI = $p")
end
